npm/vite-dev-server/src/devServer.ts

```diff
@@ -10,6 +10,7 @@ export type ViteDevServerConfig = {
   specs: Cypress.Spec[]
   cypressConfig: Cypress.PluginConfigOptions
   devServerEvents: NodeJS.EventEmitter
+  onConfigNotFound?: (devServer: 'vite', cwd: string, lookedIn: string[]) => void
```

Comment:
We probably need a `@cypress/dev-server-utils` package that defines a kind of interface that all our dev-servers must respect, which would include passing a `onConfigNotFound` error handler. We have a good idea of how future dev servers will look now, so post-10.0 we can start to make more concrete docs on how to make one.

Some things that might go there:

- `findConfig` (use `findUp`)
- common interface
- ??